testsuite/docs/optional.md

```diff
@@ -127,3 +127,43 @@ Inside of the testsuite, the scenarios that are tagged with
 @nomirror
 ```
 are executed only if you don't use a mirror.
+
+
+### Testing virtualization features
+
+Using a virtualization host with the testsuite is not mandatory.
+
+If you do not want a virtualization host minion, do not define `VIRT_SERVER_KVM` and
+`VIRT_SERVER_XEN` environment variables before you run the testsuite. That's all.
+If you want virtualization minions, make these variables point to the machines that
+will be the virtualization KVM or Xen host minions and define the `VIRT_SERVER_KVM_PASSWORD`
+and `VIRT_SERVER_XEN_PASSWORD` variables:
+
+```bash
+export VIRT_SERVER_KVM=myvirthost.example.com
+export VIRT_SERVER_KVM_PASSWORD=therootpwd
+```
+
+One of the virtualization servers can be disabled by not providing the corresponding
+environment variables.
+
+Make sure the image to use for the test virtual machines is located in
+`/var/testsuite-data/disk-image-template.qcow2` on the virtual hosts and then run
+the testsuite. In order for the virtual hosts to be able to report to the test server,
+use a bridge virtual network for the test machines.
+
+The virtual disk image should have the root file system on the `/dev/sda1` partition,
+have avahi daemon installed and running at first boot, and should be capable to be booted
+as either a Xen HVM or KVM guest. The disk images used by sumaform are good candidates
+for this.
+
+Note that the virtualization host needs to be a physical machine that needs
+to be accessible via SSH without a passphrase from the machine running the test suite. It
+also requires the `qemu-img`, `virt-install` and `guestmount` tools to be installed.
+
+Inside of the testsuite, the scenarios that are tagged with one of:
+```
+@virtualization_kvm
+@virtualization_xen
+```
+are executed only if the corresponding virtualization host minion is available.
```

Comment:
What about sumaform? I know that most of the preparation is manual, but sumaform could set up the `VIRTHOST_XEN_PASSORD` variables etc in `.bashrc`.

Sorry in advance it I missed it.

Reply:
so far I haven't touched sumaform for that

Reply:
ok